### PR TITLE
`Development`: Update naming convention in server and client coding guidelines

### PR DESCRIPTION
**Pull request by BBesrour**
#### General

- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

### Motivation and Context

As discussed in the dev meeting (13.05.2024), we need to update the naming convention for constants in the server and client where constant names should adhere to the SCREAMING_SNAKE_CASE rule. 

### Description

Updated docs

### Steps for Testing

Read the updated docs:
[server docs](https://artemis-platform--8584.org.readthedocs.build/en/8584/dev/guidelines/server.html#naming-convention)
[client docs](https://artemis-platform--8584.org.readthedocs.build/en/8584/dev/guidelines/client.html#names)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2

### CODE REVIEW

**Review by rstief — APPROVED** [review]
LGTM

**Review by marlon-luca-bu — APPROVED** [review]
Reflects the changes as discussed in the meeting 👍

File: docs/dev/guidelines/server.rst [hunk 1]
```diff
@@ -27,7 +27,27 @@ The main application is stored under ``/src/main`` and the main folders are:
 1. Naming convention
 ====================
 
-All variables, methods and classes should use CamelCase style. The only difference: the first letter of any class should be capital. Most importantly use intention-revealing, pronounceable names.
+All methods and classes should use camelCase style. The only difference: the first letter of any class should be capital. Most importantly use intention-revealing, pronounceable names.
+Variable names should also use camelCase style, where the first letter should be lowercase. For constants, i.e. arguments with the ``static final`` keywords, use all uppercase letters with underscores to separate words: SCREAMING_SNAKE_CASE.
```
Comment (florian-glombik) [inline docs/dev/guidelines/server.rst:31]:
```suggestion
All methods and classes should use camelCase style. The only difference: the first letter of any class should be capitalized. Most importantly, use intention-revealing, pronounceable names.
Variable names should also use camelCase style, where the first letter should be lowercase. For constants, i.e. arguments with the ``static final`` keywords, use all uppercase letters with underscores to separate words: SCREAMING_SNAKE_CASE.
```

File: docs/dev/guidelines/server.rst [hunk 2]
```diff
@@ -27,7 +27,27 @@ The main application is stored under ``/src/main`` and the main folders are:
...
+The only exception for this rule is for the logger, which should be named ``log``.
```
Comment (florian-glombik) [inline docs/dev/guidelines/server.rst:32]:
```suggestion
The only exception to this rule is for the logger, which should be named ``log``.
```

**Review by florian-glombik — CHANGES_REQUESTED** [review]
I've found some small mistakes in the description text - apart from these it looks good to me

**Review by Strohgelaender — APPROVED** [review]

**Review by egekurt123 — APPROVED** [review]

**Review by florian-glombik — APPROVED** [review]
Thanks for the small fixes, looks good to me now!

**Review by SimonEntholzer — APPROVED** [review]

**Review by rstief — APPROVED** [review]

**Pull request merged**